satosacontrib/perun/micro_services/perun_ensure_member.py

```diff
@@ -0,0 +1,268 @@
+        self.PARAM_REGISTRATION_URL = 'registrationUrl'
+
+        self.__logger = Logger.get_logger(self.__class__.__name__)
+
+        self.__config = config
+        self.__global_cfg = ConfigStore.get_global_cfg(
+            config["global_cfg_path"]
+        )
+        self.__attr_map_cfg = ConfigStore.get_attributes_map(
+            self.__global_cfg["attrs_cfg_path"]
+        )
+        self.__adapters_manager = AdaptersManager(
+            self.__global_cfg,
+            self.__attr_map_cfg
+        )
+
+        self.__signing_cfg = self.__global_cfg["jwk"]
+
+        if self.REGISTER_URL not in self.__config \
+                or not self.__config[self.REGISTER_URL]:
+            raise SATOSAError(
+                self.LOG_PREFIX + 'Missing configuration option \''
+                + self.REGISTER_URL + '\''
+            )
+        self.__register_url = self.__config[self.REGISTER_URL]
+
+        if self.CALLBACK_PARAMETER_NAME not in self.__config \
+                or not self.__config[self.CALLBACK_PARAMETER_NAME]:
+            raise SATOSAError(
+                self.LOG_PREFIX + 'Missing configuration option \''
+                + self.CALLBACK_PARAMETER_NAME + '\''
+            )
+        self.__callback_param_name = self.__config[self.CALLBACK_PARAMETER_NAME] # noqa e501
+
+        if self.VO_SHORT_NAME not in self.__config \
+                or not self.__config[self.VO_SHORT_NAME]:
+            raise SATOSAError(
+                self.LOG_PREFIX + 'Missing configuration option \''
+                + self.VO_SHORT_NAME + '\''
+            )
+        self.__vo_short_name = self.__config[self.VO_SHORT_NAME]
+
+        self.__group_name = self.__config[self.GROUP_NAME]
+
+        self.__unauthorized_redirect_url = \
+            self.__config["unauthorizedRedirectUrl"]
+
+        self.__registration_result_url = self.__config["registrationResultUrl"]
+
+        self.__endpoint = "/process"
+
+    def process(self, context, data):
+        """
+        This is where the micro service should modify the request / response.
+        Subclasses must call this method (or in another way make sure the
+        next callable is called).
+        @param context: The current context
+        @param data: Data to be modified
+        """
+        if data.data['perun']['user']:
+            user = data.data['perun']['user']
+        else:
+            raise SATOSAError(
+                self.LOG_PREFIX + 'Missing mandatory field '
+                '\'perun.user\' in request. Hint: Did you '
+                'configured PerunIdentity microservice '
+                'before this microservice?'
+            )
+        vo = self.__adapters_manager.get_vo(short_name=self.__vo_short_name)
+        if not vo:
+            raise SATOSAError(
+                self.LOG_PREFIX + 'VO with vo_short_name \''
+                + self.__vo_short_name + '\' not found.'
+            )
+
+        self.__handle_user(user, vo, data, context)
```

Comment:
We do not save user in data anymore. We just save user_id into attributes. Change user variables to user_id.
```suggestion
        user_id = data.attributes.get(self.__global_cfg["perun_user_id_attribute"])
        if not user_id:
            raise SATOSAError(
                self.LOG_PREFIX + 'Missing mandatory field '
                                  '\'perun.user\' in request. Hint: Did you '
                                  'configured PerunIdentity microservice '
                                  'before this microservice?'
            )
        vo = self.__adapters_manager.get_vo(short_name=self.__vo_short_name)
        if not vo:
            raise SATOSAError(
                self.LOG_PREFIX + 'VO with vo_short_name \''
                + self.__vo_short_name + '\' not found.'
            )

        self.__handle_user(user_id, vo, data, context)
```